robot/impl/local_robot.go

```diff
@@ -416,7 +416,8 @@ func newWithResources(
 
 	var ftdcWorker *ftdc.FTDC
 	if rOpts.enableFTDC {
-		ftdcWorker = ftdc.New(logger.Sublogger("ftdc"))
+		// CloudID is also known as the robot part id.
+		ftdcWorker = ftdc.New(ftdc.DefaultDirectory(config.ViamDotDir, cfg.Cloud.ID), logger.Sublogger("ftdc"))
```

Comment:
Technically has a bug -- cfg.Cloud.ID is nil for local configs. Fixed in last commit.